// koishi-plugin-webui/dist/assets/index.d69abc89.js
import{defineComponent as m,ref as w,resolveComponent as s,openBlock as i,createBlock as a,createVNode as n,createTextVNode as d,toDisplayString as g,withCtx as o,createCommentVNode as p,withDirectives as F,Fragment as f,renderList as v,vShow as E,unref as _}from"../vue.js";import{user as y,send as D,registry as z}from"../client.js";const $={class:"plugin-view"},B={class:"plugin-item"},N=d("\u7F51\u9875\u6269\u5C55"),V=d("\u526F\u4F5C\u7528"),j={class:"complexity"},A={class:"operation"},S={class:"plugin-list"};var b=m({props:{data:{type:null,required:!0}},setup(e){const u=w(!1);function h(r){const{id:t,token:l}=y.value;D("switch",{plugin:r,id:t,token:l})}return(r,t)=>{const l=s("k-badge"),C=s("k-button"),x=s("k-collapse");return i(),a("div",$,[n("div",B,[n("span",{class:["title",{"has-children":e.data.children.length}],onClick:t[1]||(t[1]=c=>e.data.children.length&&(u.value=!u.value))},[n("i",{class:["fas fa-angle-right",{show:u.value}]},null,2),d(" "+g(e.data.name)+" ",1),e.data.webExtension?(i(),a(l,{key:0,type:"default",title:"\u62E5\u6709\u7F51\u9875\u6269\u5C55\u7684\u63D2\u4EF6\u505C\u7528\u548C\u542F\u7528\u540E\u5C06\u5237\u65B0\u9875\u9762\u3002"},{default:o(()=>[N]),_:1})):p("v-if",!0),e.data.sideEffect?(i(),a(l,{key:1,type:"warning",title:"\u62E5\u6709\u526F\u4F5C\u7528\u7684\u63D2\u4EF6\u65E0\u6CD5\u88AB\u505C\u7528\u3002"},{default:o(()=>[V]),_:1})):p("v-if",!0)],2),n("span",j,g(e.data.complexity||"-"),1),n("span",A,[n(C,{class:"right",frameless:"",type:e.data.name==="webui"?"danger":"",disabled:e.data.sideEffect,onClick:t[2]||(t[2]=c=>h(e.data.id))},{default:o(()=>[d(g(e.data.complexity?"\u505C\u7528":"\u542F\u7528"),1)]),_:1},8,["type","disabled"])])]),e.data.children.length?(i(),a(x,{key:0},{default:o(()=>[F(n("div",S,[(i(!0),a(f,null,v(e.data.children,(c,k)=>(i(),a(b,{data:c,key:k},null,8,["data"]))),128))],512),[[E,u.value]])]),_:1})):p("v-if",!0)])}}}),H=`@charset "UTF-8";
:not(.has-children) > .fa-angle-right {
  font-size: 1.25rem !important;
}
:not(.has-children) > .fa-angle-right::before {
  content: "\u2022";
}
.fa-angle-right {
  margin-left: 0.75rem;
  margin-right: 0.5rem;
  transition: 0.3s ease;
  font-size: 1.25rem !important;
  width: 1.5rem;
  text-align: center;
  vertical-align: middle !important;
}
.fa-angle-right.show {
  transform: rotate(90deg);
}
.plugin-item {
  line-height: 1.6;
  user-select: none;
  padding: 0.5rem 0;
  border-bottom: #040620 1px solid;
}
.plugin-item .title {
  font-weight: bold;
}
.plugin-item .title.has-children {
  cursor: pointer;
}
.plugin-item.normal {
  color: #2477ff;
}
.plugin-item.side-effect {
  color: #F56C6C;
}
.plugin-item:hover {
  background-color: #474d8450;
}
.plugin-item .complexity {
  position: absolute;
  left: 45%;
  width: 10%;
  text-align: center;
}
.plugin-item .operation {
  float: right;
  margin-right: 3rem;
}
.plugin-view .plugin-list {
  padding-left: 2rem;
}`;const T=n("div",{class:"table-header plugin-item"},[n("span",{class:"title"},"\u63D2\u4EF6\u540D"),n("span",{class:"complexity"},"\u590D\u6742\u5EA6"),n("span",{class:"operation"},"\u64CD\u4F5C")],-1),q={class:"plugin-list root"};var L=m({setup(e){return(u,h)=>{var t;const r=s("k-card");return i(),a(r,{class:["page-plugins",{authorized:((t=_(y))==null?void 0:t.authority)>=4}],title:"\u63D2\u4EF6\u5217\u8868"},{default:o(()=>[T,n("div",q,[(i(!0),a(f,null,v(_(z),l=>(i(),a(b,{data:l},null,8,["data"]))),256))])]),_:1},8,["class"])}}}),I=`.page-plugins header {
  color: rgba(244, 244, 245, 0.6);
}
.page-plugins header span {
  transition: 0.3s ease;
}
.page-plugins header span.inactive:hover {
  cursor: pointer;
  color: rgba(244, 244, 245, 0.8);
}
.page-plugins header span:not(.inactive) {
  color: #f4f4f5;
}
.table-header {
  font-weight: bold;
  border-top: #040620 1px solid;
}
.table-header .title {
  margin-left: 3rem;
}
.page-plugins:not(.authorized) .complexity, .page-plugins:not(.authorized) .operation {
  display: none;
}`;export default L;
