// koishi-plugin-webui/dist/assets/index.dc8085e2.js
import{defineComponent as b,computed as B,ref as C,watch as x,resolveComponent as m,openBlock as n,createBlock as o,createVNode as t,toDisplayString as l,withCtx as u,createCommentVNode as g,unref as h,createTextVNode as i,Fragment as y,renderList as F}from"../vue.js";import{user as D,send as $,awesome as E}from"../client.js";const j=i("\u5B98\u65B9"),z=i("\u672C\u5730"),M=i("\u53EF\u66F4\u65B0"),N={class:"description"},V={class:"version"},W={class:"latest"},A={class:"size"},S={class:"score"},q={class:"operation"},I={key:0};var K=b({props:{data:{type:null,required:!0}},setup(e){const d=e,f=B(()=>{const{local:a,version:r}=d.data;return a&&!a.isWorkspace&&a.version!==r});function k(a){return a>=1024e3?Math.round(a/1048576)+" MB":Math.round(a/1024)+" KB"}const s=C(!1);x(()=>d.data.local,()=>{s.value=!1},{deep:!0});function _(a){const{id:r,token:c}=D.value;$("install",{name:`${a.name}@^${a.version}`,id:r,token:c}),s.value=!0}return(a,r)=>{var p,v;const c=m("k-badge"),w=m("k-button");return n(),o("tr",{class:{workspace:(p=e.data.local)==null?void 0:p.isWorkspace}},[t("td",{class:["package",e.data.local?e.data.local.isInstalled?"active":"local":"remote"]},[t("div",null,[t("a",{href:"http://npmjs.com/package/"+e.data.name,target:"blank",rel:"noopener noreferrer"},l(e.data.name.replace("koishi-plugin-","")),9,["href"]),e.data.official?(n(),o(c,{key:0,type:"success"},{default:u(()=>[j]),_:1})):g("v-if",!0),((v=e.data.local)==null?void 0:v.isWorkspace)?(n(),o(c,{key:1,type:"default"},{default:u(()=>[z]),_:1})):h(f)?(n(),o(c,{key:2,type:"warning"},{default:u(()=>[M]),_:1})):g("v-if",!0)]),t("div",N,l(e.data.description),1)],2),t("td",V,l(e.data.local?e.data.local.version:"-"),1),t("td",W,l(e.data.version),1),t("td",A,l(k(e.data.size)),1),t("td",S,l(+e.data.score.final.toFixed(2)),1),t("td",q,[s.value?(n(),o("span",I,"\u5B89\u88C5\u4E2D")):!e.data.local||h(f)?(n(),o(w,{key:1,frameless:"",onClick:r[1]||(r[1]=U=>_(e.data))},{default:u(()=>[i(l(e.data.local?"\u66F4\u65B0":"\u5B89\u88C5"),1)]),_:1})):g("v-if",!0)])],2)}}}),H=`.package {
  text-align: left;
  padding-left: 3rem;
  position: relative;
}
.package a {
  font-weight: bold;
  transition: 0.3s ease;
  color: rgba(244, 244, 245, 0.75);
}
.package a:hover {
  color: #f4f4f5;
}
.package .description {
  margin-top: 0.15rem;
  font-size: 0.9rem;
}
.package::before {
  content: "";
  position: absolute;
  border-radius: 100%;
  width: 0.5rem;
  height: 0.5rem;
  top: 50%;
  left: 1.25rem;
  transform: translateY(-50%);
  transition: 0.3s ease;
  box-shadow: 1px 1px 2px #3333;
}
.package.active::before {
  background-color: #2d8515;
}
.package.local::before {
  background-color: #e49400;
}
.package.remote::before {
  background-color: #eeeeee5f;
}
.page-awesome td:not(.package) {
  min-width: 4rem;
}
tr.workspace .latest, tr.workspace .size {
  color: rgba(244, 244, 245, 0.25);
}`;const L=t("tr",null,[t("th",null,"\u6A21\u5757\u540D\u79F0"),t("th",null,"\u5F53\u524D\u7248\u672C"),t("th",null,"\u6700\u65B0\u7248\u672C"),t("th",null,"\u603B\u4F53\u79EF"),t("th",null,"\u7EFC\u5408\u8BC4\u5206"),t("th",{class:"operation"},"\u64CD\u4F5C")],-1);var T=b({setup(e){return(d,f)=>{const k=m("k-card");return n(),o(k,{class:"page-awesome",title:"\u4F9D\u8D56\u7BA1\u7406"},{default:u(()=>[t("table",null,[L,(n(!0),o(y,null,F(h(E),s=>(n(),o(K,{data:s},null,8,["data"]))),256))])]),_:1})}}});export default T;
